test/regfox/regfox_api.integration.js

```diff
@@ -0,0 +1,83 @@
+import chai, { use, should, assert } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+
+use(chaiAsPromised);
+should();
+const { expect } = chai;
+
+import { searchRegistrations, login, getRegistrationInfo } from '../../src/regfox/regfox_api.js';
+
+const loginForTest = async () => {
+  const email = process.env.EMAIL;
+  const password = process.env.PASSWORD;
+
+  assert.exists(email);
+  assert.exists(password);
+
+  return login(email, password);
+};
+
+/* eslint-disable no-invalid-this */
+// eslint-disable-next-line valid-jsdoc
+// https://github.com/mochajs/mocha/issues/1480
+const skippableFailure = (callback) => {
+  return function () {
+    return Promise.resolve().then((...args) => {
+      return callback.apply(this, args);
+    }).catch((error) => {
+      console.error(error);
+      this.skip();
+    });
+  };
+};
+
+it.allowFail = (title, callback) => {
+  it(title, skippableFailure(callback));
+};
+
+before.allowFail = (callback) => {
+  before(skippableFailure(callback));
+};
+/* eslint-enable no-invalid-this */
+
+describe('regfox_api (integration testing)', () => {
+  it.allowFail('logs in', async () => {
+    const result = await loginForTest();
+    expect(result).to.include.keys('token');
+    expect(result.token.token).to.not.be.undefined;
+  });
+
+  describe('workflows', () => {
+    let bearerToken = undefined;
+    before.allowFail(async () => {
+      bearerToken = (await loginForTest()).token.token;
+    });
+
+    it.allowFail('searches and does not find anything', async () => {
+      assert.exists(bearerToken);
+
+      const results = await searchRegistrations('GUARANTEED_TO_BE_UNIQUE_UNLESS_SOMEONE_HAS_A_REALLY_WEIRD_NAME', bearerToken);
+      expect(results.registrants).to.be.empty;
+    });
+
+    it.allowFail('searches for a user', async () => {
+      assert.exists(bearerToken);
+
+      const results = await searchRegistrations('First Last', bearerToken);
+      expect(results.registrants).to.have.lengthOf.at.least(1);
+      const registrant = results.registrants.find((reg) => reg.status === 3 && reg.customerId === '1662788');
```

Comment:
Worth pulling these magical values out into a const of some sort listing them off as "expected to exist" values?

Reply:
Yeah, thats a good call. Let me know if this is what you had in mind or something more.

Reply:
LGTM. I'm a big fane of the [Arrange - Act - Assert](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/) test model, where you have a nice clear separation of "here's me putting my ducks in a row, here's where I do an actual test, and here's what I expected to happen". Part of the 'arrange' step is stating what your test data is going to be.

I've also written pre-test tests which do things like validate some precondition assumptions about test data that needs to exist that can't be generated on the fly, but I think that's overkill for what we're doing.

Reply:
Totally fair, I should stop being lazy and organize my test cases into AAA. :) 

---

I tried to follow [Google's practice](https://testing.googleblog.com/2008/07/tott-expect-vs-assert.html) where `assert` is used for pre-test, while `expect` is used for actual test cases. Not that mocha differentiates between them, but whatever.